core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -366,7 +391,7 @@ public void cancel(@Nullable String message, @Nullable Throwable cause) {
 
   @Override
   public void halfClose() {
-    Preconditions.checkState(stream != null, "Not started");
```

Comment:
Done. I'm cancelling the call now. However, I don't think the status will be propagated without the stream being started. Do you think we should start the stream and then do the cancelling?